tests/main/retry-network/task.yaml

```diff
@@ -0,0 +1,30 @@
+summary: Test httputil/retry.go ShouldRetryError(err)/NoNetwork(err)
+
+# ubuntu-core: no go-compiler
+systems: [-ubuntu-core-*]
+
+environment:
+    # an empty $topsrcdir/tests/go.mod seems to break importing or building go
+    # packages referenced by their import paths while under the tests directory,
+    # need to disable go modules supportfor this test
+    GO111MODULE: off
+
+prepare: |
+    go build -o detect-retry ./detect-retry.go
+
+restore: |
+    ip netns delete testns || true
+    umount /run/netns || true
+
+execute: |
+    echo "Add totally unconnected network namespace"
+    ip netns add testns
+    SNAPD_DEBUG=1 nsenter --net=/var/run/netns/testns ./detect-retry http://www.ubuntu.com > output.txt
+    #MATCH "ShouldRetryError: false" < output.txt
```

Comment:
Should it match ShouldRetryError: true? Or is it giving different result on different systems?

Reply:
I added some more docstrings that hopefully answer this question. If not please let me know and I'm happy to elaborate further :)